flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -467,7 +475,9 @@ private void createUploadFilePaths(
 
                 if (fileName.endsWith(SST_FILE_SUFFIX)) {
                     final boolean existsAlready =
-                            baseSstFiles != null && baseSstFiles.contains(stateHandleID);
+                            reuseSharedFiles
+                                    && baseSstFiles != null
+                                    && baseSstFiles.contains(stateHandleID);
```

Comment:
FYI (@rkhachatryan @dawidwys ) note, one you will have a conflict here with https://github.com/apache/flink/pull/17774